tracetools/span.go

```diff
@@ -5,25 +5,78 @@ import (
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
-	"github.com/opentracing/opentracing-go/log"
 	ddext "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 )
 
-// StartSpanFromContext will start a span from the given context with the given
-// operation name. It will also do some common/repeated setup on the span to keep
-// code a little more DRY.
-func StartSpanFromContext(ctx context.Context, operation string) (opentracing.Span, context.Context) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, operation)
-	// Make the span available for analytics in Datadog
-	span.SetTag(ddext.AnalyticsEvent, true)
-	return span, ctx
+const (
+	BackendDatadog = "datadog"
+	BackendNone    = ""
+)
+
+// StartSpanFromContext will start a span appropriate to the given tracing backend from the given context with the given
+// operation name. It will also do some common/repeated setup on the span to keep code a little more DRY.
+// If an unknown tracing backend is specified, it will return a span that noops on every operation
+func StartSpanFromContext(ctx context.Context, operation string, tracingBackend string) (Span, context.Context) {
+	switch tracingBackend {
+	case BackendDatadog:
+		span, ctx := opentracing.StartSpanFromContext(ctx, operation)
+		span.SetTag(ddext.AnalyticsEvent, true) // Make the span available for analytics in Datadog
```

Comment:
I __think__ this line might not be necessary, as we add this tag during [tracer initialisation](https://github.com/buildkite/agent/pull/1631/files#diff-f80935c46a08d736109e6791c2b0c9a1b004a9c196a8a149b3a70d5c2d3d40c0R136), but it can't hurt to be double sure it's in there i guess? ¯\_(ツ)_/¯